File: packages/client/ios/RNOta/Store/BundleStoreConfig.swift
import Foundation

/// Configuration for ``BundleStore``.
public struct BundleStoreConfig: Sendable {
  public var otaDirectory: URL
  public var installedBinaryVersion: String?
  public var supportedSchemaVersion: Int
  public var fileIo: FileIo

  public init(
    otaDirectory: URL,
    installedBinaryVersion: String? = nil,
    supportedSchemaVersion: Int = OtaPaths.supportedSchemaVersion,
    fileIo: FileIo = RealFileIo()
  ) {
    self.otaDirectory = otaDirectory
    self.installedBinaryVersion = installedBinaryVersion
    self.supportedSchemaVersion = supportedSchemaVersion
    self.fileIo = fileIo
  }
}

File: packages/client/android/src/main/java/com/rnota/store/BundleStoreConfig.kt
package com.rnota.store

import java.io.File

/**
 * Configuration for [BundleStore].
 *
 * @property otaDirectory Absolute path to `<app-data>/ota`.
 * @property installedBinaryVersion Written into state when saving (reserved for M6).
 * @property supportedSchemaVersion Must match [OtaPaths.SUPPORTED_SCHEMA_VERSION] unless testing.
 * @property fileIo Injectable IO for crash / disk-full tests. Defaults to real FS.
 */
data class BundleStoreConfig(
  val otaDirectory: File,
  val installedBinaryVersion: String? = null,
  val supportedSchemaVersion: Int = OtaPaths.SUPPORTED_SCHEMA_VERSION,
  val fileIo: FileIo = FileIo.DEFAULT,
)
